### PR TITLE
Merge generic style guide setup from Instructor center

### PR DESCRIPTION
**Pull request by trevordmiller**
# Changes

[See the full branch and updated README](https://github.com/eggheadio/egghead-io-styleguide/tree/instructor-center-merge) (easier to view together than with the PR diff).

- Take generic style guide setup out of Instructor Center and move into this repo.
- Cleanup; remove unused pieces and duplicate logic.
- Add [storybook addons](https://getstorybook.io/docs/react-storybook/addons/addon-gallery) to automate documentation.
- Update `README` with all new info.
- [Added continuous integration](https://app.codeship.com/projects/177713/configure_tests) to match.

# Result For User

storybook

---

### CODE REVIEW

**Comment by joelhooks** [comment]
nuking the "assets" isn't appropriate action. This is the only place it currently exists.

**Comment by trevordmiller** [comment]
@joelhooks @DrShpongle @tayiorbeii

I've updated this PR. Please review.

I'm very excited about this and think it turned out very nice! 🎉 I think it will make @tayiorbeii and @DrShpongle work much easier. 

Be sure to delete your `node_modules` folder, install Yarn if you don't already have it, and run `yarn` to get new dependencies.

# Next steps

## General

- The GitHub repo should be renamed from `egghead-io-styleguide` to `egghead-ui`. It would also be a good idea to rename the Codeship and Heroku projects to match for consistency.
- Change living style guide deploy location (storybook) from `styleguide.egghead.io` to `ui.egghead.io` for consistency. Not set up yet, but links to where it will be located across project.
- @tayiorbeii and @DrShpongle should update the generic components I transferred over in `src/components` to match the style guide designs; right now they are just using default Tachyons classes.
- @tayiorbeii and @DrShpongle can move the applicable pieces from the `transfer` directory to match `src/components` and `tachyons-egghead` as appropriate. I'm happy to help with this if needed, just let me know.

## Storybook

- Add Codeship continuous deployment for the storybook to ui.egghead.io. The Codeship deployment settings should run `yarn build:storybook` then deploy the built `storybook-static` directory it creates.

### Library

- Update `build:library` script to actually build `src/index.js` to `build/index.js`
- Add npm publishing for the component library. Could use [np].(https://github.com/sindresorhus/np), or [automate it with Codeship continuous deployment and semantic-release](https://www.npmjs.com/package/semantic-release-codeship-example).
- Let Trevor know it has been published and he will consume in `egghead-instructor-center`

File: README.md [hunk 1]
```diff
@@ -1,5 +1,49 @@
...
+This repo has two purposes:
+- [A component library for npm](https://www.npmjs.com/package/egghead-io-styleguide)
+- [A storybook for a living style guide with documentation](https://styleguide.egghead.io)
+
+---
+
+# Usage
+
+```
+$ yarn add 'egghead-io-styleguide'`
```
Comment (joelhooks) [inline README.md:12]: we'll want this to be `egghead-ui` but it isn't critical right now.

**Review by joelhooks — APPROVED** [review]
Looks great to me 👍

**Pull request merged**